rasa/nlu/components.py

```diff
@@ -91,13 +91,40 @@ def validate_requirements(component_names: List[Optional[Text]]) -> None:
         )
 
 
+def validate_component_keys(
+    component: "Component", component_config: Dict[Text, Any]
+) -> None:
+    """Validates that all keys for a component are valid.
+
+    Raises:
+        UserWarning: If the component has a key specified that is not used
+            by the component class, it is likely a mistake in the pipeline
+
```

Comment:
[raises is for exceptions](https://google.github.io/styleguide/pyguide.html#doc-function-raises)
```suggestion
```